src/components/dialogs/inputs/input-hooks.js

```diff
@@ -255,6 +255,35 @@ export const useButtonWithTooltip = ({ handleClick, label }) => {
     }, [label, handleClick, classes.tooltip]);
 };
 
+export const useOptionalEnumValue = (props) => {
+    const intl = useIntl();
+
+    const getEnumTranslation = useCallback(
+        (enumValue) => {
+            // translate the label of enumValue
+            const enumTranslation = props.enumObjects
+                .filter((e) => e.id === enumValue)
+                .map((e) => intl.formatMessage({ id: e.label }));
```

Comment:
i think rename  (e) by (enumElement) or (enumItem) is more expressive no ?

Reply:
ok, with enumObject then